openedx/core/djangoapps/cache_toolbox/middleware.py

```diff
@@ -141,9 +142,21 @@ def _verify_session_auth(self, request):
         auto_auth_enabled = settings.FEATURES.get('AUTOMATIC_AUTH_FOR_TESTING', False)
         if not auto_auth_enabled and hasattr(request.user, 'get_session_auth_hash'):
             session_hash = request.session.get(HASH_SESSION_KEY)
-            if not (session_hash and constant_time_compare(session_hash, request.user.get_session_auth_hash())):
-                # The session hash has changed due to a password
-                # change. Log the user out.
-                request.session.flush()
-                request.user = AnonymousUser()
-                _mark_cookie_for_deletion(request)
+            session_hash_verified = session_hash and constant_time_compare(
+                session_hash, request.user.get_session_auth_hash())
+            if session_hash_verified:
+                set_custom_attribute('CacheBackedAuthenticationMiddleware.verify.alg', settings.DEFAULT_HASHING_ALGORITHM)
```

Comment:
Minor suggestions:
1. Monitoring attributes will always be set, and aren't in a special case, so you'll have all the data you might be wondering about.
2. Returning early you can keep or ignore, but just allows for fewer if/else and might make the code a little more readable.
3. If you find a different place for `settings.DEFAULT_HASHING_ALGORITHM` custom attribute, it doesn't need to be repeated. See other PR comments.
```suggestion
            # temporary monitoring for transition to new hashing algorithm
            set_custom_attribute('DEFAULT_HASHING_ALGORITHM', settings.DEFAULT_HASHING_ALGORITHM)
            set_custom_attribute('session_hash_verified', session_hash_verified)
            if session_hash_verified:
                return
```